### PR TITLE
Feature/resizer fix

### PR DESCRIPTION
**Pull request by cmurtaugh**
This PR changes the way that we handle resizing of the Course Info Tool iframe within RCE content inside Canvas. 

Before this change we used a third-party library and a mutation observer to resize the Course Info iframes, but at some point this stopped working reliably. 

This PR changes the interaction so that the Course Info tool (inside the iframe) sends a `postMessage` to the parent window indicating that its content has loaded and the iframe height can be safely updated. The `postMessage` also includes the height of the iframed content. 

On the receiving end, we install an event listener that watches for the `postMessage`; if it gets a message from an acceptable origin and in an acceptable format, it will resize the iframe to the height supplied in the message. 

Note that this new functionality depends on an update to the javascript file in the global Canvas theme; the two changes should be released together.

### CODE REVIEW

File: course_info/templates/course_info/widget.html [hunk 1]
```diff
@@ -4,11 +4,21 @@
...
 {% endblock head_extra %}
 {% block extra_js %}
-  <script src="{% static 'js/iframeResizer.contentWindow.min.js' %}"></script>
+  {% if referer %}
+    <script>
+    $(document).ready(function() {
+      h = document.body.offsetHeight;
+      data = {'message': 'ready_for_resize', 'height': h};
+      window.parent.postMessage(data, '{{ referer }}');
+      console.log('child frame ready_for_resize to '+ h);
```
Comment (sapnamysore) [inline course_info/templates/course_info/widget.html]: Can this be cleaned up after testing?

**Comment by sapnamysore** [comment]
Looks good👍!  can we  reproduce and test this issue with and without the fix?

File: course_info/templates/course_info/widget.html (hunk 1 above)
Comment (cmurtaugh) [reply course_info/templates/course_info/widget.html]: yeah - we can always re-enable it later if needed

**Comment by cmurtaugh** [comment]
I made a bunch more changes to get this to be testable on dev (where the REST API is returning numbers as floats instead of ints at the moment). 

The dev instance of this tool should be installable on any canvas.dev course; you can see it in action on:
* https://canvas.dev.tlt.harvard.edu/courses/5806
* https://canvas.dev.tlt.harvard.edu/courses/5498
* https://canvas.dev.tlt.harvard.edu/courses/3761

**Comment by cmsnowden** [comment]
Just one pretty minor thing. It's not super obvious that you can't edit the list of attributes to be displayed. A user may click the 'i', and by unclicking things, expect that the unclicked things will disappear from the present display. Instead, what happens is another instance of the iframe gets inserted according to what is being clicked. If they happen to unclick everything, thinking that will get rid of everything, we end up with a 200px high empty space. Sizing the default to something less than 200px height might help mitigate a bit anyway. Removing the unwanted iframe completely requires a bit of not-super-friendly maneuvering by the user.

**Comment by cmurtaugh** [comment]
@cmsnowden - one reason I wanted to keep the initial height at 200px is that the frame doesn't get resized to fit the content when in editing mode, so if it's too small most of the content gets cut off and users might not be able to see enough of it.  I agree that they could get themselves into a weird state by unchecking everything, but hopefully that's not a common occurrence.

File: course_info/views.py [hunk 2]
```diff
@@ -174,27 +174,31 @@ def widget(request):
...
 def editor(request):
+    _logger.debug('EDITOR: {}'.format(request.POST))
     course_instance_id = request.POST.get('lis_course_offering_sourcedid')
+
     course_context = _course_context(request, _ORDERED_FIELD_NAMES, True,
                                      course_instance_id=course_instance_id)
     course_context['launch_presentation_return_url'] = \
         request.POST.get('launch_presentation_return_url')
+    course_context['canvas_course_id'] = request.POST.get('custom_canvas_course_id')
     return render(request, 'course_info/editor.html', course_context)
```
Comment (cmsnowden) [inline course_info/views.py:222]: Looks good to me.
Tested it on dev and looks good.

**Comment by Chris-Thornton-Harvard** [comment]
👍Looks good

**Pull request closed without merging**